artifactor/__init__.py

```diff
@@ -237,6 +243,48 @@ def parse_setup_dir(test_name, test_location, artifactor_config, artifact_dir, r
     return {'artifact_path': path}, None
 
 
+def parse_build_dir(artifactor_config, artifact_dir, run_id):
+    """
+    Convenience fire_hook for built in hook
+    """
+    run_type = artifactor_config.get('per_run')
+    overwrite = artifactor_config.get('reuse_dir', False)
+    report_path = setup_report_dir(root_dir=artifact_dir, run_type=run_type, run_id=run_id,
+                                   overwrite=overwrite)
+    return {'report_path': report_path}, None
+
+
+def setup_report_dir(root_dir=None, run_type=None, run_id=None, overwrite=True):
+    """
+    Sets up the artifact dir and returns it.
+    """
+    orig_path = os.path.abspath(root_dir)
+
+    if run_id:
+        run_id = str(run_id)
+
+    if run_type == "run" and run_id:
+        path = os.path.join(orig_path, run_id)
+    elif run_type == "test" and run_id:
+        path = os.path.join(orig_path, run_id)
+    else:
+        path = os.path.join(orig_path)
+
+    try:
+        os.makedirs(path)
```

Comment:
**optional** add a note to switch to pathlib2 and exists_ok once we go for python3 only